### PR TITLE
make online work with sklearn pipelines

### PR DESCRIPTION
**Pull request by davegri**
# Brainflow
* All brainflow related code is now in `board.py` where there is a `Board` class. This class implements `__enter__` and `__exit__` methods so that we can use the [Python "with" statement](https://www.geeksforgeeks.org/with-statement-in-python/) to automatically stop recording when the scope is exited
* `Board` is instantiated with a `use_synthetic` flag which can be true or false. `recording.py` takes this parameter from `recording_params.json` .
* All brainflow related constants are also now in `board.py`
* Unified online and offline, recording @haggaishapira I know I was initially against this, but I've reconsidered given the amount of code duplication.
# Psychopy
* `win.update` is deprecated in _psychopy_, so we should only use `win.flip`. I've also added a method `show_stim_for_duration` for repeating the same four actions when we want to display a stimulus for some specified duration: _draw, flip, wait, flip_
* recording is now started only after pressing any key, and can be stopped at any time by pressing `Esc`

# Pipeline
* All of the code now uses sklearns Pipeline object. In short, a Pipeline includes all of our transformations (preprocessing, feature extraction) + prediction, and can be created based on a set of hyperparameters that define each stage.

### CODE REVIEW

**Comment by haggaishapira** [comment]
does the online predictor still guess idle for every input?

**Comment by davegri** [comment]
> does the online predictor still guess idle for every input?

Unfortunately yes, I need to do more investigating but my initial instinct is that it's not a bug but just a bad model, also I've only tested with a synthetic board

File: live_stream.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+from board import Board
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.signal
+import scipy.stats
+import mne
+
+
+def health_check():
```
Comment (idopongo) [inline live_stream.py]: better to call the file and main function with the same name

File: live_stream.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            data = get_next_data(board, window_size)
+            errors_by_chan = check_chan_health(data)
+            update_chan_plots(chan_plots, data, window_size)
+            update_montage_plot(ax, montage_plot, errors_by_chan, chan_error_texts)
+            plt.draw()
+            plt.pause(1e-3)
+
+
+def on_press(event):
+    if event.key == 'escape':
```
Comment (idopongo) [inline live_stream.py]: should be a constant

File: recording.py [hunk 3]
```diff
@@ -40,80 +41,87 @@ def load_params():
...
-        sleep(params["get_ready_duration"])
+    with Board(use_synthetic=params["use_synthetic_board"]) as board:
+        for i, marker in enumerate(trial_markers):
+            show_stim_for_duration(win, progress_text(win, i + 1, len(trial_markers), marker),
+                                   params["get_ready_duration"])
+            core.wait(params["calibration_duration"])
+            board.insert_marker(marker)
+            show_stim_for_duration(win, marker_stim(win, marker), params["trial_duration"])
+
+            if pipeline:
```
Comment (idopongo) [inline recording.py:66]: fucking awsooome!
love the design

**Review by idopongo — APPROVED** [review]

**Pull request merged**